utils/workers/Creators.py

```diff
@@ -0,0 +1,364 @@
+    creator: CreateWorker = None
+    converter: Converters.Converter = None
+    labels: dict = None
+
+
+
+    def __post_init__(self):
+
+        self.creator = self.creator(
+            orig_dir=self.orig_dir,
+            roi_dir=self.roi_dir,
+            output_dir=self.output_dir,
+            base_file_name=self.base_file_name,
+            combine=self.combine,
+            bitmask=self.bitmask,
+            converter=self.converter
+        )
+
+    def create(self, ohifviewer_info):
+        labels = self.creator.create(ohifviewer_info)
+        return labels
+
+    def get_affine(self):
+        return self.creator.get_affine()
+
+
+class DicomCreator(CreateWorker):
+    def __init__(self, orig_dir, roi_dir, output_dir, base_file_name, combine, bitmask, converter):
+        super().__init__(orig_dir, roi_dir, output_dir, base_file_name, combine, bitmask, converter)
+        self.dicoms = {}
+        self.max_labels = 31
+
+    def create(self, ohifviewer_info):
+        labels = self.get_labels(ohifviewer_info)
+        self.get_dicoms()
+        self.make_data(labels, ohifviewer_info)
+        return labels
+
+
+    def generate_name(self, label, combine):
+        if combine:
+            label_out = "ALL"
+        else:
+            label_out = re.sub("[^0-9a-zA-Z]+", "_", label)
+
+        output_filename = "ROI_" + label_out + "_" + self.base_file_name
+
+        if output_filename.endswith(".gz"):
+            output_filename = output_filename[: -1 * len(".nii.gz")]
+        elif output_filename.endswith(".nii"):
+            output_filename = output_filename[: -1 * len(".nii")]
+        elif output_filename.endswith(".dicom.zip"):
+            output_filename = output_filename[: -1 * len(".dicom.zip")]
+        elif output_filename.endswith(".zip"):
+            output_filename = output_filename[: -1 * len(".zip")]
+
+        return output_filename
+
+
+    def get_dicoms(self):
+        # Acquire ROI data
+        globdir = self.orig_dir / "*"
+        dicom_files = glob.glob(globdir.as_posix())
+        dicom_files.sort()
+        dicom_files = [Path(d) for d in dicom_files]
+        dicoms = {d: pydicom.read_file(d) for d in dicom_files}
+        example_dicom = dicoms[dicom_files[0]]
+        self.shape = [
+            example_dicom.pixel_array.shape[0],
+            example_dicom.pixel_array.shape[1],
+            len(dicom_files),
+        ]
+        self.dicoms = dicoms
+
+    def get_affine(self):
+
+        positions = np.mat(
+            [loaded_dicom.ImagePositionPatient for dpath,loaded_dicom in self.dicoms.items()]
+        )
+        zdif = np.diff(positions, axis=0)[:, 0]
+        mzdif = stats.mode(zdif).mode[0][0]
+        mzdif = np.round(mzdif, 4)
+        ds = list(self.dicoms.values())[0]
+        affine = np.mat(
+            [
+                [1 * mzdif, 0, 0],
+                [0, 1 * ds.PixelSpacing[0], 0],
+                [0, 0, 1 * ds.PixelSpacing[1]],
+            ]
+        )
```

Comment:
I'm not sure I'm understanding this, wouldn't the affine just be the `ImageOrientationPatient`, i.e.:
```python
affine = np.eye(3)
affine[:,0] = dcm.ImageOrientationPatient[:3]
affinep[:,1] = dcm.ImageOrientationPatient[3:]
```
Or is this normal to the slice?

Reply:
Maybe affine isn't the best name.  This is giving us a way to calculate volume.  It's x/y/z dimensions in the plane of acquisition.